server/events/project_command_builder.go

```diff
@@ -372,6 +379,64 @@ func (p *DefaultProjectCommandBuilder) buildProjectPlanCommand(ctx *command.Cont
 	}
 
 	var pcc []command.ProjectContext
+
+	// use the default repository workspace because it is the only one guaranteed to have an atlantis.yaml,
+	// other workspaces will not have the file if they are using pre_workflow_hooks to generate it dynamically
+	defaultRepoDir, err := p.WorkingDir.GetWorkingDir(ctx.Pull.BaseRepo, ctx.Pull, DefaultWorkspace)
+	if err != nil {
+		return pcc, err
+	}
+
+	if p.StrictPlanFileList {
+		modifiedFiles, err := p.VCSClient.GetModifiedFiles(ctx.Pull.BaseRepo, ctx.Pull)
```

Comment:
I believe we call this function a lot. Is it possible that this function was already called and we can pass that information from where it was originally called so it's available here?

Reply:
I've based this code on the `buildPlanAllCommands` function. Both `buildPlanAllCommands` and `buildProjectPlanCommand` are called from `BuildPlanCommands` which is called from `plan_command_runner.run`.

From what I can see both `buildPlanAllCommands` and `buildProjectPlanCommand` are the first functions to call the `GetModifiedFiles` in this stack, adding something before would probably hurt more than help since right now `buildPlanAllCommands` only uses this when the `RestrictFileList` is true and adding an "always on" `GetModifiedFiles` before that would end up adding unnecessary requests when `RestrictFileList` is false.

Reply:
Okay so it sounds like you're saying an additional call here to the VCS is necessary because we're calling it in the other functions but not in the root calling function making it difficult to pass through to this function?

Reply:
I was only calling attention to this to ensure that this was thought about. We're always hoping to not increase the number of our api calls if we can help it.

Reply:
Yes, you got it right, at least in this stack this is the first time that the VCS is necessary so it's unavoidable to call it there. I totally agree that we should do only the minimum necessary API calls.